# DSCI 100 Final Project
## Group 33

### Introduction

#### Background information:
A Computer Science research team at UBC set up a Minecraft server, recording player's actions to explore how people play video games. We wanted to figure out which player characteristics or behaviours could be used to predict their subscription status to the game-related newsletter.

#### Question:
Can player’s hours played, age  and experience level predict whether a player is subscribed to the game-related newsletter?

#### Data Set
The data set used to answer this question was the Player’s Dataset (players.csv). The dataset contains 196 observations and 7 variables, including player's experience, age, gender, email, name, subscription status and hours played. However the variables of interest for answering this question are age, experience and hours played.

### Methods & Results

In [7]:
library(tidyverse)
library(repr)
library(tidymodels) # Load the packages
player_data<-read_csv("data/players.csv") # loads data

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [8]:
player_data<- select(player_data,subscribe,played_hours,Age,experience) # select needed variable
player_data$Age[is.na(player_data$Age)] <- median(player_data$Age, na.rm = TRUE) # add median for NA age data


#### performs a summary of the data set that is relevant for exploratory data analysis related to the planned analysis 

In [9]:
head(player_data,5) 
summary(player_data) #Generate summary

subscribe played_hours Age experience
1  TRUE     30.3          9  Pro       
2  TRUE      3.8         17  Veteran   
3 FALSE      0.0         17  Veteran   
4  TRUE      0.7         21  Amateur   
5  TRUE      0.1         21  Regular

 subscribe        played_hours          Age         experience       
 Mode :logical   Min.   :  0.000   Min.   : 8.00   Length:196        
 FALSE:52        1st Qu.:  0.000   1st Qu.:17.00   Class :character  
 TRUE :144       Median :  0.100   Median :19.00   Mode  :character  
                 Mean   :  5.846   Mean   :20.51                     
                 3rd Qu.:  0.600   3rd Qu.:22.00                     
                 Max.   :223.100   Max.   :50.00                     

**Table: Summary of numeric varibles**
| Variable          | Class      | Min     | 1st Quartile | Median | Mean  | 3rd Quartile | Max    | NA's |
|------------------|-----------|--------|--------|--------|--------|--------|--------|------|
| **played_hours** | double    | 0.000  | 0.000  | 0.100  | 5.846  | 0.600  | 223.100 | 0    |
| **Age**        | double     | 8.00   | 17.00  | 19.00  | 20.52  | 22.00  | 50.00  | 2    |

### Discussion

### References